pulsar-broker/src/main/java/org/apache/pulsar/client/api/RawReader.java

```diff
@@ -49,6 +51,13 @@
      */
     CompletableFuture<RawMessage> readNextAsync();
 
+    /**
+     * Acknowledge all messages as read up until <i>messageId</i>.
+     * @param messageId to cumulatively acknowledge to
+     * @param properties a map of properties which will be stored with the acknowledgement
+     */
+    CompletableFuture<Void> acknowledgeAsync(MessageId messageId, Map<String,Long> properties);
```

Comment:
This is a cumulative ack. I think it would be better to call acknowledgeCumulative rather than acknowledge. Just to keep the concepts/names consistent across the project.

Reply:
will change.

Reply:
I was also thinking I should just rename RawReader to CompactionReader, since acknowledging with properties is very much a compaction thing rather than a raw reader thing.